### PR TITLE
chore: modify AgentBaker windows test

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind improvement

**What this PR does / why we need it**:
1. Clarify workflow titles.
2. Support 23H2 in e2e pipeline.
3. Only use 2022 gen1 and 23H2 gen2 as default to reduce cost.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: .pipelines/templates/e2e-windows-template.yaml [hunk 1]
```diff
@@ -53,7 +55,8 @@ jobs:
           export WINDOWS_PACKAGE_VERSION=$KUBERNETES_VERSION
         fi
         export K8S_VERSION=${WINDOWS_PACKAGE_VERSION//./}
-        export WINDOWS_E2E_STORAGE_CONTAINER=${{ parameters.storageAccount }}-$K8S_VERSION
```
Comment (AbelHu) [inline .pipelines/templates/e2e-windows-template.yaml:56]: Does the old code not work?

Comment (junjiezhang1997) [reply .pipelines/templates/e2e-windows-template.yaml:56]: “23H2” contains an uppercase letter, but a valid storage account name shouldn't contain uppercase letters. So I use "declare -l" to change it to lowercase letters.

File: .pipelines/templates/e2e-windows-template.yaml [hunk 2]
```diff
@@ -53,7 +55,8 @@ jobs:
...
+        declare -l WINDOWS_E2E_STORAGE_CONTAINER=${{ parameters.storageAccount }}-$K8S_VERSION
```
Comment (AbelHu) [inline .pipelines/templates/e2e-windows-template.yaml:58]: Do we cleanup the storage account and containers?

Comment (AbelHu) [reply .pipelines/templates/e2e-windows-template.yaml:58]: I found some code but it only touches ws2019
    if [ "$WINDOWS_E2E_IMAGE" == "2019-containerd" ]; then
        cleanupOutdatedFiles
    fi

Comment (junjiezhang1997) [reply .pipelines/templates/e2e-windows-template.yaml:58]: We don't cleanup the storage account. We just cleanup the containers when the related k8s version is not supported any more in https://github.com/Azure/AgentBaker/blob/master/e2e/windows/e2e-helper.sh#L19

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**